doubly_linkedlist.py

```diff
@@ -26,14 +29,17 @@ def append(self, val):
                 if current.next == None:
                     if current == self.head:
                         self.head.next = Node(val, None, self.head)
+                        self.tail = self.head.next
```

Comment:
The whole point of having a tail is to avoid needing to loop to get to the tail end.